GitUp/Application/Document.m

```diff
@@ -936,7 +942,99 @@ - (void)_enterQuickViewWithHistoryCommit:(GCHistoryCommit*)commit commitList:(NS
   }
   
   _quickViewController.commit = commit;
-  [self _setWindowMode:kWindowModeString_Map_QuickView];
+  
+  [self _animateQuickViewForCommit:commit appearing:YES transition:^{
+    [self _setWindowMode:kWindowModeString_Map_QuickView];
+  }];
+}
+
+- (void)_animateQuickViewForCommit:(GCHistoryCommit*)commit appearing:(BOOL)appearing transition:(void (^)(void))transitionBlock {
+  [CATransaction flush];
+  [CATransaction begin];
+  [CATransaction setDisableActions:YES];
+  
+  BOOL animationInFlight = YES;
+  if (!_fixedSnapshotLayer) {
+    animationInFlight = NO;
+    _fixedSnapshotLayer = [CALayer layer];
+    _fixedSnapshotLayer.backgroundColor = _mainWindow.backgroundColor.CGColor;
+    _animatingSnapshotLayer = [CALayer layer];
+    _animatingSnapshotLayer.backgroundColor = _mainWindow.backgroundColor.CGColor;
+    _animatingSnapshotLayer.shadowOpacity = 0.2;
+    _animatingSnapshotLayer.shadowRadius = 30;
+  }
+  
+  if (animationInFlight) {
+    transitionBlock();
+  } else if (appearing) {
+    _fixedSnapshotLayer.contents = [_contentView snapshotImage];
+    transitionBlock();
+    _animatingSnapshotLayer.contents = [_contentView snapshotImage];
+  } else {
+    _animatingSnapshotLayer.contents = [_contentView snapshotImage];
+    transitionBlock();
+    _fixedSnapshotLayer.contents = [_contentView snapshotImage];
+  }
+  
+  [_contentView.layer addSublayer:_fixedSnapshotLayer];
+  [_contentView.layer addSublayer:_animatingSnapshotLayer];
+  
+  _fixedSnapshotLayer.frame = _contentView.layer.bounds;
+  
+  CALayer* fromLayer;
+  if (animationInFlight) {
+    fromLayer = _animatingSnapshotLayer.presentationLayer ?: _animatingSnapshotLayer;
```

Comment:
Please don't use the shortcut version of the ternary operator as I don't use it elsewhere in the code base.